docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,460 @@
+Detail will be discussed in [upcoming](#configure-the-model) section.
+
+```c
+// Create dataset
+ml_train_dataset_h dataset;
+ml_train_dataset_create_with_generator(&dataset, generator_train_cb, generator_valid_cb, generator_test_cb);
+
+// configure dataset
+ml_train_dataset_set_property(dataset, "buffer_size=100", NULL);
+
+// after set dataset to model,
+// you do not need to destroy dataset since ownership is transferred to the model.
+ml_train_model_set_dataset(model, dataset);
+```
+
+## Construct a Model
+
+A model can be constructed with `ml_train_model_construct`.
+If you have a file that describes the model, the file can be used to construct initially with `ml_train_model_construct_with_file`.
+Even if the model is constructed from a file, Switching/modifying/setting a component is possible until compiling with `ml_train_model_compile`.
+
+### Construct a Model from the Description File
+
+As of now, only INI formatted files `*.ini` is supported to construct a model from a file.
+
+#### Create a Model from INI Formatted File
+
+Special sections `[Model]` and `[Dataset]` are respectively referring to `model` and `dataset` object.
+Rest of INI sections map to a `layer`. Keys and values from each section set properties of the layer.
+All keys and values are treated as case-insensitive.
+
+For example,
+
+```ini
+[Model] # Special section that describes model itself
+Type = NeuralNetwork  # Model Type : only NeuralNetwork is supported as of now
+Optimizer = adam  # Optimizer : sgd (stochastic gradient decent),
+          #             adam (Adaptive Moment Estimation)
+#### optimizer related properties
+Learning_rate = 0.0001  # Learning Rate for the optimizer
+Decay_rate = 0.96 # for the decay_rate for the decayed learning rate
+Decay_steps = 1000       # decay step for the exponential decayed learning rate
+beta1 = 0.9     # beta 1 for adam
+beta2 = 0.9999  # beta 2 for adam
+epsilon = 1e-7  # epsilon for adam
+
+#### Model compile related properties
+batch_size = 9
+loss = cross      # Cost(loss) function : mse (mean squared error)
+                        #                       cross (cross entropy)
+
+####  Model run related properties
+Epochs = 20     # Epochs
+save_path = "model.bin"   # model path to save / read parameters
+
+[DataSet] # Special section that describes dataset.
+BufferSize=9
+TrainData="trainingSet.dat"
+ValidData="validationSet.dat"
+LabelData="label.dat"
+
+# Layer Sections, each section name refers to name of the layer
+[inputlayer]
+Type = input
+Input_Shape = 1:1:62720 # Input Layer Dimension channel:height:width
+Normalization = true
+
+[outputlayer]
+Type = fully_connected
+Unit = 2    # Output Layer Dimension ( = Weight Width )
+bias_initializer = zeros
+weight_initializer = xavier_uniform
+Activation = sigmoid  # activation : sigmoid, softmax
+Weight_Decay = l2norm
+weight_Decay_Lambda = 0.005
+```
+
+Some restrictions must be kept:
```

Comment:
The following restrictions must be adhered to:

Reply:
Done